.dotnet/tests/Moderations/ModerationsMockTests.cs

```diff
@@ -0,0 +1,695 @@
+﻿using System;
+using System.ClientModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using OpenAI.Moderations;
+using OpenAI.Tests.Utility;
+
+namespace OpenAI.Tests.Moderations;
+
+[TestFixture(true)]
+[TestFixture(false)]
+[Parallelizable(ParallelScope.All)]
+[Category("Smoke")]
+public partial class ModerationsMockTests : SyncAsyncTestBase
```

Comment:
Avoided adding live testing to moderations because we would need to come up with not-so-friendly prompts for many of the tests.